File: src/main/java/org/kowboy/command/AbstractCommandExecutor.java
package org.kowboy.command;

import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.plugin.java.JavaPlugin;

import static org.kowboy.util.ChatUtils.sendHelp;

public abstract class AbstractCommandExecutor implements CommandExecutor {
    protected abstract String getDescription();
    protected abstract String getUsage();
    protected final JavaPlugin plugin;

    protected AbstractCommandExecutor(JavaPlugin plugin) {
        this.plugin = plugin;
    }

    protected void help(CommandSender sender, boolean withDescription) {
        if (withDescription) {
            sendHelp(sender, "Description: " + getDescription());
        }
        sendHelp(sender, "Usage: " + getUsage());
    }
}

File: src/main/kotlin/org/kowboy/command/AbstractCommandExecutor.kt
package org.kowboy.command

import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.plugin.java.JavaPlugin
import org.kowboy.util.ChatUtils.sendHelp

abstract class AbstractCommandExecutor protected constructor(val plugin: JavaPlugin) : CommandExecutor {
    protected abstract val description: String
    protected abstract val usage: String
    protected fun help(sender: CommandSender, withDescription: Boolean) {
        if (withDescription) {
            sendHelp(sender, "Description: $description")
        }
        sendHelp(sender, "Usage: $usage")
    }

}